cni/pkg/install-cni/pkg/install/server.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"net/http"
+	"sync/atomic"
+
+	"istio.io/istio/cni/pkg/install-cni/pkg/constants"
+	"istio.io/pkg/log"
+)
+
+// StartServer initializes and starts a web server that exposes liveliness and readiness endpoints at port 8000.
+func StartServer() *atomic.Value {
+	router := http.NewServeMux()
+	isReady := initRouter(router)
+
+	go func() {
+		log.Fatala(http.ListenAndServe(":"+constants.Port, router))
```

Comment:
Won't this crash when the server exits? do we want that?

Reply:
K8S should kill it if the healthz endpoint dies, right? So having the process die and possibly restart before the pod errors out seems sane.